### DISCUSSION TITLE
Which flows are supported?

### DISCUSSION THREAD

**Discussion by raman-nbg** [ory/hydra#2891] [Q&A] [answered]
I couldn't find a list of supported/implemented flows by hydra in the docs. Can you tell me which flows are supported, please?

**Comment by vinckr** [comment, marked as answer]
Hello @raman-nbg 
All flows according the [OAuth2.0 spec](https://datatracker.ietf.org/doc/html/rfc6749) and [OpenIDConnect](https://openid.net/developers/specs/) are implemented.

We do however discourage the use of the Resource Owner Password Credentials Flow, read more here https://www.ory.sh/oauth2-for-mobile-app-spa-browser/.

May I ask what is the use case you are looking to solve with Ory Hydra?

**Reply by raman-nbg** [reply]
Thank you for the quick response.

I'm just wondering if hydra also supports the CIBA flow. This might be something that we need in the future (but not sure yet). I just wanted to check if this is possible. But still I'm not sure if this is already possible with hydra because this is quite new. Can you confirm that CIBA is already implemented in hydra?

**Reply by vinckr** [reply]
Sure!
At the moment we don't support it. It is of course possible to implement and there has been some community interest already: https://github.com/ory/hydra/issues/2004, but the issue has been closed due to low interest. 
We don't have concrete plans to implement it, but if there is community interest we will of course consider it. 
As always contributions are also very welcome 😄 .

**Reply by raman-nbg** [reply]
For now we also don't need it very urgently. If we need it in the future, I would contribute to implement this. Thank you.
